src/main/java/seedu/address/model/client/Timezone.java

```diff
@@ -17,21 +22,31 @@
 public class Timezone {
 
     public static final String GMT_STRING = "GMT";
-    public static final int SMALLEST_NEGATIVE_OFFSET = 12;
-    public static final int LARGEST_POSITIVE_OFFSET = 14;
 
-    public static final String MESSAGE_CONSTRAINTS =
-            "Timezone should be in the form \"" + GMT_STRING + "+X\" or \"" + GMT_STRING
-            + "-X\" where X is a number.\n" + "Largest offset is " + GMT_STRING
-            + "+" + LARGEST_POSITIVE_OFFSET + " and smallest offset is " + GMT_STRING
-            + "-" + SMALLEST_NEGATIVE_OFFSET;
+    public static final Set<String> VALID_TIMEZONES = new HashSet<>();
+
+    static {
+        Set<String> allZones = ZoneId.getAvailableZoneIds();
+        List<String> zoneList = new ArrayList<>(allZones);
+        for (String s : zoneList) {
+            ZoneOffset offset = LocalDateTime.now().atZone(ZoneId.of(s)).getOffset();
+            String out = String.format("%s", offset);
+            VALID_TIMEZONES.add(out);
+        }
+        VALID_TIMEZONES.remove("Z");
+        VALID_TIMEZONES.add("+00:00");
+    }
+
+    public static final String MESSAGE_CONSTRAINTS = "Timezone should be in the form \"" + GMT_STRING + "+HH:MM\" or \""
+            + GMT_STRING + "-HH:MM\" where HH is the offset in hours and MM is the offset in minutes. The full list of "
+            + "valid timezones can be found at https://www.timeanddate.com/time/current-number-time-zones.html";
```

Comment:
same comment as in UG about the link. Is there any better links?

Reply:
can this be used instead? https://greenwichmeantime.com/time-gadgets/time-zone-converter/